htdocs/admin/content/messages.php

```diff
@@ -1,822 +1,714 @@
+    }
+
+    echo "<form action=\"?page=$page&sub=$sub\" method=\"POST\">";
+    echo "<table width=\"300\" class=\"tb\">";
+    echo "<tr>
+        <th width=\"50\">Sender:</th>
+        <td>";
+    $playerUser = $userRepository->getUser($cu->playerId);
+    if ($playerUser !== null)
+    {
+        echo "<input type=\"radio\" name=\"from_id\" id=\"from_id_1\" value=\"".$cu->playerId."\" checked=\"checked\" /> <label for=\"from_id_1\">".$playerUser->nick." (InGame-Account #".$playerUser->id.")</label><br/>";
+        echo "<input type=\"radio\" name=\"from_id\" id=\"from_id_0\" value=\"0\" /> <label for=\"from_id_0\">System</label><br/>";
+    }
+    else {
+        echo "System <input type=\"hidden\" name=\"from_id\" value=\"0\" />";
+    }
+    echo "</td></tr>";
+    echo "<tr>
+        <th>Empfänger:</th>
+        <td class=\"tbldata\" width=\"250\">
+        <b>An:</b>
+        <input type=\"radio\" name=\"rcpt_type\" id=\"rcpt_type_1\" value=\"1\"  checked=\"checked\"  onclick=\"document.getElementById('message_user_to').style.display='none';\" /> <label for=\"rcpt_type_1\">Alle Spieler</label>
+        <input type=\"radio\" name=\"rcpt_type\" id=\"rcpt_type_0\" value=\"0\"  onclick=\"document.getElementById('message_user_to').style.display='';\" /> <label for=\"rcpt_type_0\">Einzelner Empfänger</label>
+        <select name=\"message_user_to\" id=\"message_user_to\" style=\"display:none\">";
+        $userNicks = $userRepository->getUserNicknames();
+        foreach ($userNicks as $userId => $userNick) {
+            echo "<option value=\"".$userId."\"";
+            echo ">".$userNick."</option>";
+        }
+    echo "</select> &nbsp;
+
+    <br/>
+    <b>Typ:</b>
+    <input type=\"radio\" name=\"msg_type\" value=\"0\" id=\"msg_type_0\"  checked=\"checked\" /> <label for=\"msg_type_0\">InGame-Nachricht</label>
+    <input type=\"radio\" name=\"msg_type\" value=\"1\" id=\"msg_type_1\" /> <label for=\"msg_type_1\">E-Mail</label>
+    <input type=\"radio\" name=\"msg_type\" value=\"2\" id=\"msg_type_2\" /> <label for=\"msg_type_2\">InGame-Nachricht &amp; E-Mail</label>
+    </td></tr>";
+    echo "<tr>
+        <th>Betreff:</th>
+        <td><input type=\"text\" name=\"message_subject\" value=\"".$subj."\" size=\"60\" maxlength=\"255\"></td></tr>";
+    echo "<tr>
+        <th>Text:</th>
+        <td><textarea name=\"message_text\" rows=\"10\" cols=\"60\">".$text."</textarea></td></tr>";
+    echo "</table>";
+    echo "<p align=\"center\"><input type=\"submit\" class=\"button\" name=\"submit\" value=\"Senden\"></p>";
+    echo "</form>";
+}
+
+function manageReports(Request $request, ReportRepository $reportRepository, UserRepository $userRepository): void
+{
+    global $page;
+    global $sub;
+
+    echo "<h1>Berichte</h1>";
+    //
+    // Suchresultate
+    //
+    if ($request->request->has('user_search') && $_POST['user_search']!="" || $request->query->get('action') == "searchresults")
+    {
+        $sql = '';
```

Comment:
use the repository (or at least the dbQuery builder)

Reply:
I would like to tackle the report part in a separate branch/PR

Reply:
yes please :)